packages/cspell-lib/config/cspell-default.json

```diff
@@ -11,33 +11,132 @@
-        { "languageId": "map",    "enabled": false },
-        { "languageId": "image",  "enabled": false },
-        { "languageId": "binary", "enabled": false },
-
-        { "languageId": "markdown,html,mdx",                   "ignoreRegExpList": ["/&[a-z]+;/g"] },
-        { "languageId": "html",                                "ignoreRegExpList": ["href"] },
+        {
+            "languageId": "*",
+            "dictionaries": [
+                "companies",
+                "softwareTerms",
+                "filetypes"
+            ]
+        },
+        {
+            "languageId": "csharp",
+            "allowCompoundWords": true,
+            "dictionaries": [
+                "csharp",
+                "dotnet",
+                "npm"
+            ]
+        },
+        {
+            "languageId": "javascript,javascriptreact",
+            "dictionaries": [
+                "typescript",
+                "node",
+                "npm"
+            ]
+        },
+        {
+            "languageId": "typescript,typescriptreact,mdx",
+            "dictionaries": [
+                "typescript",
+                "node",
+                "npm"
+            ]
+        },
+        {
+            "languageId": "javascriptreact,typescriptreact,mdx",
+            "dictionaries": [
+                "html",
+                "html-symbol-entities",
+                "css",
+                "fonts"
+            ]
+        },
+        {
+            "languageId": "markdown,asciidoc",
+            "dictionaries": [
+                "npm",
+                "html",
+                "html-symbol-entities"
+            ]
+        },
+        {
+            "languageId": "html,pug,jade,php,handlebars",
+            "dictionaries": [
+                "html",
+                "fonts",
+                "typescript",
+                "css",
+                "npm",
+                "html-symbol-entities"
+            ]
+        },
+        {
+            "languageId": "json,jsonc",
+            "dictionaries": [
+                "node",
+                "npm"
+            ]
+        },
+        {
+            "languageId": "php",
+            "dictionaries": [
+                "php"
+            ]
+        },
+        {
+            "languageId": "css,less,scss",
+            "dictionaries": [
+                "fonts",
+                "css"
+            ]
+        },
+        {
+            "languageId": "map",
+            "enabled": false
+        },
+        {
+            "languageId": "image",
+            "enabled": false
+        },
+        {
+            "languageId": "binary",
+            "enabled": false
+        },
+        {
+            "languageId": "markdown,html,mdx",
+            "ignoreRegExpList": [
+                "/&[a-z]+;/g"
+            ]
+        },
+        {
+            "languageId": "html",
+            "ignoreRegExpList": [
+                "href",
+                "id=\"(?:[^\\\"]+|\\.)*\"",
```

Comment:
would you define these as patterns in [html/cspell-ext.json](https://github.com/streetsidesoftware/cspell-dicts/blob/master/packages/html/cspell-ext.json) like found in [python/cspell-ext.json](https://github.com/streetsidesoftware/cspell-dicts/blob/master/packages/python/cspell-ext.json)?

Then then can be turned on/off and even overridden in case they do not perform so well.

Reply:
OK, I also wasn't sure if these should be a named regex like "href"

Reply:
Using named patterns has some advantages.
1. It becomes easier to know what it should do (assuming it is a good name).
1. It can be overridden easily with an improved pattern.

Reply:
OK, I can see it being a combine regex, maybe called "IDRef" or something similar, as that's what these represent. Do you have a preference to doing that here and using the named regex in this place, or just shifting it to cspell-dicts?

Reply:
I would rather have them defined in cspell-dict/packages/html/cspell-ext.json. That way they can be changed independently from the tool.